src/python/pants/core/goals/lint.py

```diff
@@ -215,7 +215,7 @@ async def lint(
     file_requests = tuple(
         request_type(specs_snapshot.snapshot.files)
         for request_type in file_request_types
-        if request_type.name in specified_names
+        if request_type.name in specified_names and specs_snapshot.snapshot.files
     )
```

Comment:
Peformance nit: you can make this O(1) rather than O(n):

```suggestion
        if request_type.name in specified_names
    ) if specs_snapshot.snapshot.files else ()
```

(Black might not like this formatting, so better to manually apply the fix)